reactor-core/src/main/java/reactor/core/publisher/MonoCompletionStage.java

```diff
@@ -56,6 +56,12 @@ public void subscribe(CoreSubscriber<? super T> actual) {
         }
 
         future.whenComplete((v, e) -> {
+            if (sds.isCancelled()) {
```

Comment:
should test if error is not null ?

Reply:
Mmh, that was the source of the issue in the first place. if the mono is cancelled, sometimes the user wants to also cancel the `Future`, but said cancellation triggers `whenComplete`...

If that sounds very important that a cancelled Mono should still receive an error, then _maybe_ we can forward any error *other than a `CancellationException`* 🤔